Source/objects.cpp

```diff
@@ -1345,14 +1342,35 @@ void AddTrap(Object &trap)
 	trap._oVar4 = 0;
 }
 
-void AddObjectLight(Object &object, int r)
+void AddObjectLight(Object &object)
 {
-	if (ApplyObjectLighting) {
-		DoLighting(object.position, r, {});
-		object._oVar1 = -1;
-	} else {
-		object._oVar1 = 0;
+	int radius;
+	switch (object._otype) {
+	case OBJ_STORYCANDLE:
+	case OBJ_L5CANDLE:
+		radius = 3;
+		break;
+	case OBJ_L1LIGHT:
+	case OBJ_SKFIRE:
+	case OBJ_CANDLE1:
+	case OBJ_CANDLE2:
+	case OBJ_BOOKCANDLE:
+	case OBJ_BCROSS:
+	case OBJ_TBCROSS:
+		radius = 5;
+		break;
+	case OBJ_TORCHL:
+	case OBJ_TORCHR:
+	case OBJ_TORCHL2:
+	case OBJ_TORCHR2:
+		radius = 8;
+		break;
+	default:
+		return;
 	}
+
+	DoLighting(object.position, radius, {});

```

Comment:
Question:
`DoLighting` expects that `LoadingMapObjects` is true (else the light would be removed when updating these tiles).
But `LoadingMapObjects` is not always true when calling `AddObjectLight`.
Examples: `InitObjects()` and `CreateThemeRooms()` when loading a game.
Is it relevant?

Reply:
>  DoLighting expects that LoadingMapObjects is true (else the light would be removed when updating these tiles).

This is only true during gameplay, while initially generating the level lighting is applied to `dLIght`, when the level is fully generated `dLight` will be copied to `dPreLight`. So the global should be off during level generation, but on during gameplay.

Reply:
Forgot about this one.